src/sql/platform/connection/common/connectionConfig.ts

```diff
@@ -241,6 +241,11 @@ export class ConnectionConfig {
 				profile.id = generateUuid();
 				changed = true;
 			}
+			// SSMS 19 requires "user", fix any profiles created without user property.
+			if (profile.providerName === 'MSSQL' && !profile.options.user) {
```

Comment:
should this strictly check for `profile.options.user === undefined`? The current check of `!profile.options.user` will also evaluate to true for an empty string